lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +31,51 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self.json_files = list(self.path_to_raw_txt_data.glob('*_meta.json'))
+        self.txt_files = list(self.path_to_raw_txt_data.glob('*_raw.txt'))
+        self._validate_dataset()
+        self._storage = {}
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError('file does not exist')
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError('path does not lead to directory')
+        if not self.json_files or not self.txt_files:
+            raise EmptyDirectoryError('directory is empty')
+        len_json, len_txt = len(self.json_files), len(self.txt_files)
+        if len_json != len_txt:
+            raise InconsistentDatasetError('number of meta and raw files is not equal')
+        for json, txt in zip(self.json_files, self.txt_files):
+            if not json.stat().st_size or not txt.stat().st_size:
+                raise InconsistentDatasetError("files are empty")
+        self.json_files = list(map(lambda link: link.name, self.json_files))
+        self.txt_files = list(map(lambda link: link.name, self.txt_files))
+        get_nums = lambda name: int(re.match(r'\d*', name).group(0))
+        json_nums = list(map(get_nums, self.json_files))
+        txt_nums = list(map(get_nums, self.txt_files))
+        if len_json != max(json_nums) or len_txt != max(txt_nums):
+            raise InconsistentDatasetError("files' IDs contain slips")
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry
         """
+        for id, file in enumerate(self.txt_files, 1):
+            article = Article(file, id)
```

Comment:
no need to create instance - from_raw already creates an instance

Reply:
Done